### ISSUE TITLE
error after running pip install confluent-kafka

### ISSUE THREAD

**Issue by 1182986209** [confluentinc/confluent-kafka-python#194]
I have already done the Prerequisites:
Python >= 2.7 or Python 3.x
librdkafka >= 0.9.1 
But I got error after running pip install confluent-kafka:
Downloading/unpacking confluent-kafka
  Downloading confluent-kafka-0.9.4.tar.gz (40kB): 40kB downloaded
  Running setup.py (path:/tmp/pip_build_eadeegf/confluent-kafka/setup.py) egg_info for package confluent-kafka
    
Installing collected packages: confluent-kafka
  Running setup.py install for confluent-kafka
    building 'confluent_kafka.cimpl' extension
    gcc -pthread -fno-strict-aliasing -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/home/eadeegf/tmp/python/include/python2.7 -c confluent_kafka/src/confluent_kafka.c -o build/temp.linux-x86_64-2.7/confluent_kafka/src/confluent_kafka.o
    In file included from confluent_kafka/src/confluent_kafka.c:17:
    confluent_kafka/src/confluent_kafka.h:21:32: error: librdkafka/rdkafka.h: No such file or directory
    In file included from confluent_kafka/src/confluent_kafka.c:17:
    confluent_kafka/src/confluent_kafka.h:96: error: expected \u2018)\u2019 before \u2018err\u2019
    confluent_kafka/src/confluent_kafka.h:97: error: expected \u2018)\u2019 before \u2018err\u2019
    confluent_kafka/src/confluent_kafka.h:123: error: expected specifier-qualifier-list before \u2018rd_kafka_t\u2019
    confluent_kafka/src/confluent_kafka.h:223: error: expected \u2018=\u2019, \u2018,\u2019, \u2018;\u2019, \u2018asm\u2019 or \u2018__attribute__\u2019 before \u2018*\u2019 token
    confluent_kafka/src/confluent_kafka.h:227: warning: type defaults to \u2018int\u2019 in declaration of \u2018rd_kafka_topic_partition_list_t\u2019
    confluent_kafka/src/confluent_kafka.h:227: error: expected \u2018;\u2019, \u2018,\u2019 or \u2018)\u2019 before \u2018*\u2019 token
    confluent_kafka/src/confluent_kafka.h:228: error: expected \u2018=\u2019, \u2018,\u2019, \u2018;\u2019, \u2018asm\u2019 or \u2018__attribute__\u2019 before \u2018*\u2019 token
    confluent_kafka/src/confluent_kafka.h:253: error: expected specifier-qualifier-list before \u2018rd_kafka_timestamp_type_t\u2019
    confluent_kafka/src/confluent_kafka.h:258: warning: type defaults to \u2018int\u2019 in declaration of \u2018rd_kafka_message_t\u2019
    confluent_kafka/src/confluent_kafka.h:258: error: expected \u2018;\u2019, \u2018,\u2019 or \u2018)\u2019 before \u2018*\u2019 token
    confluent_kafka/src/confluent_kafka.h:275: warning: type defaults to \u2018int\u2019 in declaration of \u2018rd_kafka_topic_t\u2019
    confluent_kafka/src/confluent_kafka.h:275: error: expected \u2018;\u2019, \u2018,\u2019 or \u2018)\u2019 before \u2018*\u2019 token
    confluent_kafka/src/confluent_kafka.c:52: error: expected specifier-qualifier-list before \u2018rd_kafka_resp_err_t\u2019
    confluent_kafka/src/confluent_kafka.c: In function \u2018KafkaError_code\u2019:
    confluent_kafka/src/confluent_kafka.c:60: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019
    confluent_kafka/src/confluent_kafka.c: In function \u2018KafkaError_str\u2019:
    confluent_kafka/src/confluent_kafka.c:64: error: \u2018KafkaError\u2019 has no member named \u2018str\u2019
    confluent_kafka/src/confluent_kafka.c:65: error: \u2018KafkaError\u2019 has no member named \u2018str\u2019
    confluent_kafka/src/confluent_kafka.c:67: warning: implicit declaration of function \u2018rd_kafka_err2str\u2019
    confluent_kafka/src/confluent_kafka.c:67: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019
    confluent_kafka/src/confluent_kafka.c: In function \u2018KafkaError_name\u2019:
    confluent_kafka/src/confluent_kafka.c:72: warning: implicit declaration of function \u2018rd_kafka_err2name\u2019
    confluent_kafka/src/confluent_kafka.c:72: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019
    confluent_kafka/src/confluent_kafka.c: In function \u2018KafkaError_dealloc\u2019:
    confluent_kafka/src/confluent_kafka.c:105: error: \u2018KafkaError\u2019 has no member named \u2018str\u2019
    confluent_kafka/src/confluent_kafka.c:106: error: \u2018KafkaError\u2019 has no member named \u2018str\u2019
    confluent_kafka/src/confluent_kafka.c: In function \u2018KafkaError_str0\u2019:
    confluent_kafka/src/confluent_kafka.c:112: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019
    confluent_kafka/src/confluent_kafka.c:112: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019
    confluent_kafka/src/confluent_kafka.c:112: error: \u2018KafkaError\u2019 has no member named \u2018str\u2019
    confluent_kafka/src/confluent_kafka.c:112: error: \u2018KafkaError\u2019 has no member named \u2018str\u2019
    confluent_kafka/src/confluent_kafka.c:112: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019
    confluent_kafka/src/confluent_kafka.c: In function \u2018KafkaError_hash\u2019:
    confluent_kafka/src/confluent_kafka.c:120: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019
    confluent_kafka/src/confluent_kafka.c: In function \u2018KafkaError_richcompare\u2019:
    confluent_kafka/src/confluent_kafka.c:132: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019
    confluent_kafka/src/confluent_kafka.c:139: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019
    confluent_kafka/src/confluent_kafka.c:142: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019
    confluent_kafka/src/confluent_kafka.c:145: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019
    confluent_kafka/src/confluent_kafka.c:148: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019
    confluent_kafka/src/confluent_kafka.c:151: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019
    confluent_kafka/src/confluent_kafka.c:154: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019
    confluent_kafka/src/confluent_kafka.c: At top level:
    confluent_kafka/src/confluent_kafka.c:221: error: expected declaration specifiers or \u2018...\u2019 before \u2018rd_kafka_resp_err_t\u2019
    confluent_kafka/src/confluent_kafka.c: In function \u2018KafkaError_init\u2019:
    confluent_kafka/src/confluent_kafka.c:222: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019
    confluent_kafka/src/confluent_kafka.c:222: error: \u2018code\u2019 undeclared (first use in this function)
    confluent_kafka/src/confluent_kafka.c:222: error: (Each undeclared identifier is reported only once
    confluent_kafka/src/confluent_kafka.c:222: error: for each function it appears in.)
    confluent_kafka/src/confluent_kafka.c:224: error: \u2018KafkaError\u2019 has no member named \u2018str\u2019
    confluent_kafka/src/confluent_kafka.c:226: error: \u2018KafkaError\u2019 has no member named \u2018str\u2019
    confluent_kafka/src/confluent_kafka.c: At top level:
    confluent_kafka/src/confluent_kafka.c:232: error: expected \u2018)\u2019 before \u2018err\u2019
    confluent_kafka/src/confluent_kafka.c:258: error: expected \u2018)\u2019 before \u2018err\u2019
    confluent_kafka/src/confluent_kafka.c: In function \u2018Message_partition\u2019:
    confluent_kafka/src/confluent_kafka.c:312: error: \u2018RD_KAFKA_PARTITION_UA\u2019 undeclared (first use in this function)
    confluent_kafka/src/confluent_kafka.c: In function \u2018Message_timestamp\u2019:
    confluent_kafka/src/confluent_kafka.c:329: error: \u2018Message\u2019 has no member named \u2018tstype\u2019
    confluent_kafka/src/confluent_kafka.c: At top level:
    confluent_kafka/src/confluent_kafka.c:532: warning: type defaults to \u2018int\u2019 in declaration of \u2018rd_kafka_message_t\u2019
    confluent_kafka/src/confluent_kafka.c:532: error: expected \u2018;\u2019, \u2018,\u2019 or \u2018)\u2019 before \u2018*\u2019 token
    confluent_kafka/src/confluent_kafka.c:588: error: expected declaration specifiers or \u2018...\u2019 before \u2018rd_kafka_resp_err_t\u2019
    confluent_kafka/src/confluent_kafka.c: In function \u2018TopicPartition_setup\u2019:
    confluent_kafka/src/confluent_kafka.c:592: warning: implicit declaration of function \u2018KafkaError_new_or_None\u2019
    confluent_kafka/src/confluent_kafka.c:592: error: \u2018err\u2019 undeclared (first use in this function)
    confluent_kafka/src/confluent_kafka.c: In function \u2018TopicPartition_init\u2019:
    confluent_kafka/src/confluent_kafka.c:608: error: \u2018RD_KAFKA_PARTITION_UA\u2019 undeclared (first use in this function)
    confluent_kafka/src/confluent_kafka.c:609: error: \u2018RD_KAFKA_OFFSET_INVALID\u2019 undeclared (first use in this function)
    confluent_kafka/src/confluent_kafka.c:620: error: too many arguments to function \u2018TopicPartition_setup\u2019
    confluent_kafka/src/confluent_kafka.c: At top level:
    confluent_kafka/src/confluent_kafka.c:797: error: expected declaration specifiers or \u2018...\u2019 before \u2018rd_kafka_resp_err_t\u2019
    confluent_kafka/src/confluent_kafka.c: In function \u2018TopicPartition_new0\u2019:
    confluent_kafka/src/confluent_kafka.c:803: error: \u2018err\u2019 undeclared (first use in this function)
    confluent_kafka/src/confluent_kafka.c:803: error: too many arguments to function \u2018TopicPartition_setup\u2019
    confluent_kafka/src/confluent_kafka.c: At top level:
    confluent_kafka/src/confluent_kafka.c:816: warning: type defaults to \u2018int\u2019 in declaration of \u2018rd_kafka_topic_partition_list_t\u2019
    confluent_kafka/src/confluent_kafka.c:816: error: expected \u2018;\u2019, \u2018,\u2019 or \u2018)\u2019 before \u2018*\u2019 token
    confluent_kafka/src/confluent_kafka.c:839: error: expected \u2018=\u2019, \u2018,\u2019, \u2018;\u2019, \u2018asm\u2019 or \u2018__attribute__\u2019 before \u2018*\u2019 token
    confluent_kafka/src/confluent_kafka.c:883: error: expected \u2018)\u2019 before \u2018*\u2019 token
    confluent_kafka/src/confluent_kafka.c:909: error: expected \u2018)\u2019 before \u2018*\u2019 token
    confluent_kafka/src/confluent_kafka.c: In function \u2018Handle_clear\u2019:
    confluent_kafka/src/confluent_kafka.c:952: error: \u2018Handle\u2019 has no member named \u2018error_cb\u2019
    confluent_kafka/src/confluent_kafka.c:953: error: \u2018Handle\u2019 has no member named \u2018error_cb\u2019
    confluent_kafka/src/confluent_kafka.c:953: error: \u2018Handle\u2019 has no member named \u2018error_cb\u2019
    confluent_kafka/src/confluent_kafka.c:953: error: \u2018Handle\u2019 has no member named \u2018error_cb\u2019
    confluent_kafka/src/confluent_kafka.c:955: error: \u2018Handle\u2019 has no member named \u2018stats_cb\u2019
    confluent_kafka/src/confluent_kafka.c:956: error: \u2018Handle\u2019 has no member named \u2018stats_cb\u2019
    confluent_kafka/src/confluent_kafka.c:956: error: \u2018Handle\u2019 has no member named \u2018stats_cb\u2019
    confluent_kafka/src/confluent_kafka.c:956: error: \u2018Handle\u2019 has no member named \u2018stats_cb\u2019
    confluent_kafka/src/confluent_kafka.c:958: error: \u2018Handle\u2019 has no member named \u2018initiated\u2019
    confluent_kafka/src/confluent_kafka.c:959: error: \u2018Handle\u2019 has no member named \u2018tlskey\u2019
    confluent_kafka/src/confluent_kafka.c: In function \u2018Handle_traverse\u2019:
    confluent_kafka/src/confluent_kafka.c:966: error: \u2018Handle\u2019 has no member named \u2018error_cb\u2019
    confluent_kafka/src/confluent_kafka.c:967: error: \u2018Handle\u2019 has no member named \u2018error_cb\u2019
    confluent_kafka/src/confluent_kafka.c:967: error: \u2018Handle\u2019 has no member named \u2018error_cb\u2019
    confluent_kafka/src/confluent_kafka.c:969: error: \u2018Handle\u2019 has no member named \u2018stats_cb\u2019
    confluent_kafka/src/confluent_kafka.c:970: error: \u2018Handle\u2019 has no member named \u2018stats_cb\u2019
    confluent_kafka/src/confluent_kafka.c:970: error: \u2018Handle\u2019 has no member named \u2018stats_cb\u2019
    confluent_kafka/src/confluent_kafka.c: At top level:
    confluent_kafka/src/confluent_kafka.c:982: error: expected \u2018)\u2019 before \u2018*\u2019 token
    confluent_kafka/src/confluent_kafka.c:1042: error: expected declaration specifiers or \u2018...\u2019 before \u2018rd_kafka_conf_t\u2019
    confluent_kafka/src/confluent_kafka.c:1043: error: expected declaration specifiers or \u2018...\u2019 before \u2018rd_kafka_topic_conf_t\u2019
    confluent_kafka/src/confluent_kafka.c: In function \u2018producer_conf_set_special\u2019:
    confluent_kafka/src/confluent_kafka.c:1050: warning: implicit declaration of function \u2018KafkaError_new0\u2019
    confluent_kafka/src/confluent_kafka.c:1050: error: \u2018RD_KAFKA_RESP_ERR__INVALID_ARG\u2019 undeclared (first use in this function)
    confluent_kafka/src/confluent_kafka.c:1057: error: \u2018Handle\u2019 has no member named \u2018u\u2019
    confluent_kafka/src/confluent_kafka.c:1058: error: \u2018Handle\u2019 has no member named \u2018u\u2019
    confluent_kafka/src/confluent_kafka.c:1071: warning: implicit declaration of function \u2018rd_kafka_topic_conf_set_partitioner_cb\u2019
    confluent_kafka/src/confluent_kafka.c:1072: error: \u2018tconf\u2019 undeclared (first use in this function)
    confluent_kafka/src/confluent_kafka.c:1072: error: \u2018rd_kafka_msg_partitioner_random\u2019 undeclared (first use in this function)
    confluent_kafka/src/confluent_kafka.c:1075: error: \u2018rd_kafka_msg_partitioner_consistent\u2019 undeclared (first use in this function)
    confluent_kafka/src/confluent_kafka.c:1078: error: \u2018rd_kafka_msg_partitioner_consistent_random\u2019 undeclared (first use in this function)
    confluent_kafka/src/confluent_kafka.c:1104: error: \u2018RD_KAFKA_RESP_ERR__NOT_IMPLEMENTED\u2019 undeclared (first use in this function)
    confluent_kafka/src/confluent_kafka.c:1110: error: \u2018Handle\u2019 has no member named \u2018u\u2019
    confluent_kafka/src/confluent_kafka.c:1111: error: \u2018Handle\u2019 has no member named \u2018u\u2019
    confluent_kafka/src/confluent_kafka.c:1111: error: \u2018Handle\u2019 has no member named \u2018u\u2019
    confluent_kafka/src/confluent_kafka.c:1111: error: \u2018Handle\u2019 has no member named \u2018u\u2019
    confluent_kafka/src/confluent_kafka.c:1113: error: \u2018Handle\u2019 has no member named \u2018u\u2019
    confluent_kafka/src/confluent_kafka.c:1114: error: \u2018Handle\u2019 has no member named \u2018u\u2019
    confluent_kafka/src/confluent_kafka.c:1118: error: \u2018Producer_partitioner_cb\u2019 undeclared (first use in this function)
    confluent_kafka/src/confluent_kafka.c: At top level:
    confluent_kafka/src/confluent_kafka.c:1133: error: expected declaration specifiers or \u2018...\u2019 before \u2018rd_kafka_conf_t\u2019
    confluent_kafka/src/confluent_kafka.c:1134: error: expected declaration specifiers or \u2018...\u2019 before \u2018rd_kafka_topic_conf_t\u2019
    confluent_kafka/src/confluent_kafka.c: In function \u2018consumer_conf_set_special\u2019:
    confluent_kafka/src/confluent_kafka.c:1139: error: \u2018RD_KAFKA_RESP_ERR__INVALID_ARG\u2019 undeclared (first use in this function)
    confluent_kafka/src/confluent_kafka.c:1146: error: \u2018Handle\u2019 has no member named \u2018u\u2019
    confluent_kafka/src/confluent_kafka.c:1147: error: \u2018Handle\u2019 has no member named \u2018u\u2019
    confluent_kafka/src/confluent_kafka.c: At top level:
    confluent_kafka/src/confluent_kafka.c:1162: error: expected \u2018=\u2019, \u2018,\u2019, \u2018;\u2019, \u2018asm\u2019 or \u2018__attribute__\u2019 before \u2018*\u2019 token
    confluent_kafka/src/confluent_kafka.c: In function \u2018CallState_begin\u2019:
    confluent_kafka/src/confluent_kafka.c:1335: error: \u2018Handle\u2019 has no member named \u2018tlskey\u2019
    confluent_kafka/src/confluent_kafka.c: In function \u2018CallState_end\u2019:
    confluent_kafka/src/confluent_kafka.c:1343: error: \u2018Handle\u2019 has no member named \u2018tlskey\u2019
    confluent_kafka/src/confluent_kafka.c: In function \u2018CallState_get\u2019:
    confluent_kafka/src/confluent_kafka.c:1358: error: \u2018Handle\u2019 has no member named \u2018tlskey\u2019
    confluent_kafka/src/confluent_kafka.c: In function \u2018libversion\u2019:
    confluent_kafka/src/confluent_kafka.c:1396: warning: implicit declaration of function \u2018rd_kafka_version_str\u2019
    confluent_kafka/src/confluent_kafka.c:1397: warning: implicit declaration of function \u2018rd_kafka_version\u2019
    confluent_kafka/src/confluent_kafka.c: In function \u2018KafkaError_add_errs\u2019:
    confluent_kafka/src/confluent_kafka.c:1438: warning: implicit declaration of function \u2018rd_kafka_get_err_descs\u2019
    confluent_kafka/src/confluent_kafka.c:1474: error: invalid use of undefined type \u2018struct rd_kafka_err_desc\u2019
    confluent_kafka/src/confluent_kafka.c:1474: error: dereferencing pointer to incomplete type
    confluent_kafka/src/confluent_kafka.c:1477: error: invalid use of undefined type \u2018struct rd_kafka_err_desc\u2019
    confluent_kafka/src/confluent_kafka.c:1477: error: dereferencing pointer to incomplete type
    confluent_kafka/src/confluent_kafka.c:1479: error: invalid use of undefined type \u2018struct rd_kafka_err_desc\u2019
    confluent_kafka/src/confluent_kafka.c:1479: error: dereferencing pointer to incomplete type
    confluent_kafka/src/confluent_kafka.c:1483: error: invalid use of undefined type \u2018struct rd_kafka_err_desc\u2019
    confluent_kafka/src/confluent_kafka.c:1483: error: dereferencing pointer to incomplete type
    confluent_kafka/src/confluent_kafka.c:1483: error: invalid use of undefined type \u2018struct rd_kafka_err_desc\u2019
    confluent_kafka/src/confluent_kafka.c:1483: error: dereferencing pointer to incomplete type
    confluent_kafka/src/confluent_kafka.c: In function \u2018_init_cimpl\u2019:
    confluent_kafka/src/confluent_kafka.c:1561: error: \u2018RD_KAFKA_TIMESTAMP_NOT_AVAILABLE\u2019 undeclared (first use in this function)
    confluent_kafka/src/confluent_kafka.c:1562: error: \u2018RD_KAFKA_TIMESTAMP_CREATE_TIME\u2019 undeclared (first use in this function)
    confluent_kafka/src/confluent_kafka.c:1563: error: \u2018RD_KAFKA_TIMESTAMP_LOG_APPEND_TIME\u2019 undeclared (first use in this function)
    confluent_kafka/src/confluent_kafka.c:1565: error: \u2018RD_KAFKA_OFFSET_BEGINNING\u2019 undeclared (first use in this function)
    confluent_kafka/src/confluent_kafka.c:1566: error: \u2018RD_KAFKA_OFFSET_END\u2019 undeclared (first use in this function)
    confluent_kafka/src/confluent_kafka.c:1567: error: \u2018RD_KAFKA_OFFSET_STORED\u2019 undeclared (first use in this function)
    confluent_kafka/src/confluent_kafka.c:1568: error: \u2018RD_KAFKA_OFFSET_INVALID\u2019 undeclared (first use in this function)
    error: command 'gcc' failed with exit status 1
    Complete output from command /home/eadeegf/tmp/python/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip_build_eadeegf/confluent-kafka/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-zJROpL-record/install-record.txt --single-version-externally-managed --compile:
    running install

running build

running build_py

creating build

creating build/lib.linux-x86_64-2.7

creating build/lib.linux-x86_64-2.7/confluent_kafka

copying confluent_kafka/__init__.py -> build/lib.linux-x86_64-2.7/confluent_kafka

creating build/lib.linux-x86_64-2.7/confluent_kafka/avro

copying confluent_kafka/avro/cached_schema_registry_client.py -> build/lib.linux-x86_64-2.7/confluent_kafka/avro

copying confluent_kafka/avro/__init__.py -> build/lib.linux-x86_64-2.7/confluent_kafka/avro

creating build/lib.linux-x86_64-2.7/confluent_kafka/kafkatest

copying confluent_kafka/kafkatest/verifiable_client.py -> build/lib.linux-x86_64-2.7/confluent_kafka/kafkatest

copying confluent_kafka/kafkatest/verifiable_consumer.py -> build/lib.linux-x86_64-2.7/confluent_kafka/kafkatest

copying confluent_kafka/kafkatest/verifiable_producer.py -> build/lib.linux-x86_64-2.7/confluent_kafka/kafkatest

copying confluent_kafka/kafkatest/__init__.py -> build/lib.linux-x86_64-2.7/confluent_kafka/kafkatest

creating build/lib.linux-x86_64-2.7/confluent_kafka/avro/serializer

copying confluent_kafka/avro/serializer/__init__.py -> build/lib.linux-x86_64-2.7/confluent_kafka/avro/serializer

copying confluent_kafka/avro/serializer/message_serializer.py -> build/lib.linux-x86_64-2.7/confluent_kafka/avro/serializer

running build_ext

building 'confluent_kafka.cimpl' extension

creating build/temp.linux-x86_64-2.7

creating build/temp.linux-x86_64-2.7/confluent_kafka

creating build/temp.linux-x86_64-2.7/confluent_kafka/src

gcc -pthread -fno-strict-aliasing -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/home/eadeegf/tmp/python/include/python2.7 -c confluent_kafka/src/confluent_kafka.c -o build/temp.linux-x86_64-2.7/confluent_kafka/src/confluent_kafka.o

In file included from confluent_kafka/src/confluent_kafka.c:17:

confluent_kafka/src/confluent_kafka.h:21:32: error: librdkafka/rdkafka.h: No such file or directory

In file included from confluent_kafka/src/confluent_kafka.c:17:

confluent_kafka/src/confluent_kafka.h:96: error: expected \u2018)\u2019 before \u2018err\u2019

confluent_kafka/src/confluent_kafka.h:97: error: expected \u2018)\u2019 before \u2018err\u2019

confluent_kafka/src/confluent_kafka.h:123: error: expected specifier-qualifier-list before \u2018rd_kafka_t\u2019

confluent_kafka/src/confluent_kafka.h:223: error: expected \u2018=\u2019, \u2018,\u2019, \u2018;\u2019, \u2018asm\u2019 or \u2018__attribute__\u2019 before \u2018*\u2019 token

confluent_kafka/src/confluent_kafka.h:227: warning: type defaults to \u2018int\u2019 in declaration of \u2018rd_kafka_topic_partition_list_t\u2019

confluent_kafka/src/confluent_kafka.h:227: error: expected \u2018;\u2019, \u2018,\u2019 or \u2018)\u2019 before \u2018*\u2019 token

confluent_kafka/src/confluent_kafka.h:228: error: expected \u2018=\u2019, \u2018,\u2019, \u2018;\u2019, \u2018asm\u2019 or \u2018__attribute__\u2019 before \u2018*\u2019 token

confluent_kafka/src/confluent_kafka.h:253: error: expected specifier-qualifier-list before \u2018rd_kafka_timestamp_type_t\u2019

confluent_kafka/src/confluent_kafka.h:258: warning: type defaults to \u2018int\u2019 in declaration of \u2018rd_kafka_message_t\u2019

confluent_kafka/src/confluent_kafka.h:258: error: expected \u2018;\u2019, \u2018,\u2019 or \u2018)\u2019 before \u2018*\u2019 token

confluent_kafka/src/confluent_kafka.h:275: warning: type defaults to \u2018int\u2019 in declaration of \u2018rd_kafka_topic_t\u2019

confluent_kafka/src/confluent_kafka.h:275: error: expected \u2018;\u2019, \u2018,\u2019 or \u2018)\u2019 before \u2018*\u2019 token

confluent_kafka/src/confluent_kafka.c:52: error: expected specifier-qualifier-list before \u2018rd_kafka_resp_err_t\u2019

confluent_kafka/src/confluent_kafka.c: In function \u2018KafkaError_code\u2019:

confluent_kafka/src/confluent_kafka.c:60: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019

confluent_kafka/src/confluent_kafka.c: In function \u2018KafkaError_str\u2019:

confluent_kafka/src/confluent_kafka.c:64: error: \u2018KafkaError\u2019 has no member named \u2018str\u2019

confluent_kafka/src/confluent_kafka.c:65: error: \u2018KafkaError\u2019 has no member named \u2018str\u2019

confluent_kafka/src/confluent_kafka.c:67: warning: implicit declaration of function \u2018rd_kafka_err2str\u2019

confluent_kafka/src/confluent_kafka.c:67: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019

confluent_kafka/src/confluent_kafka.c: In function \u2018KafkaError_name\u2019:

confluent_kafka/src/confluent_kafka.c:72: warning: implicit declaration of function \u2018rd_kafka_err2name\u2019

confluent_kafka/src/confluent_kafka.c:72: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019

confluent_kafka/src/confluent_kafka.c: In function \u2018KafkaError_dealloc\u2019:

confluent_kafka/src/confluent_kafka.c:105: error: \u2018KafkaError\u2019 has no member named \u2018str\u2019

confluent_kafka/src/confluent_kafka.c:106: error: \u2018KafkaError\u2019 has no member named \u2018str\u2019

confluent_kafka/src/confluent_kafka.c: In function \u2018KafkaError_str0\u2019:

confluent_kafka/src/confluent_kafka.c:112: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019

confluent_kafka/src/confluent_kafka.c:112: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019

confluent_kafka/src/confluent_kafka.c:112: error: \u2018KafkaError\u2019 has no member named \u2018str\u2019

confluent_kafka/src/confluent_kafka.c:112: error: \u2018KafkaError\u2019 has no member named \u2018str\u2019

confluent_kafka/src/confluent_kafka.c:112: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019

confluent_kafka/src/confluent_kafka.c: In function \u2018KafkaError_hash\u2019:

confluent_kafka/src/confluent_kafka.c:120: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019

confluent_kafka/src/confluent_kafka.c: In function \u2018KafkaError_richcompare\u2019:

confluent_kafka/src/confluent_kafka.c:132: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019

confluent_kafka/src/confluent_kafka.c:139: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019

confluent_kafka/src/confluent_kafka.c:142: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019

confluent_kafka/src/confluent_kafka.c:145: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019

confluent_kafka/src/confluent_kafka.c:148: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019

confluent_kafka/src/confluent_kafka.c:151: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019

confluent_kafka/src/confluent_kafka.c:154: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019

confluent_kafka/src/confluent_kafka.c: At top level:

confluent_kafka/src/confluent_kafka.c:221: error: expected declaration specifiers or \u2018...\u2019 before \u2018rd_kafka_resp_err_t\u2019

confluent_kafka/src/confluent_kafka.c: In function \u2018KafkaError_init\u2019:

confluent_kafka/src/confluent_kafka.c:222: error: \u2018KafkaError\u2019 has no member named \u2018code\u2019

confluent_kafka/src/confluent_kafka.c:222: error: \u2018code\u2019 undeclared (first use in this function)

confluent_kafka/src/confluent_kafka.c:222: error: (Each undeclared identifier is reported only once

confluent_kafka/src/confluent_kafka.c:222: error: for each function it appears in.)

confluent_kafka/src/confluent_kafka.c:224: error: \u2018KafkaError\u2019 has no member named \u2018str\u2019

confluent_kafka/src/confluent_kafka.c:226: error: \u2018KafkaError\u2019 has no member named \u2018str\u2019

confluent_kafka/src/confluent_kafka.c: At top level:

confluent_kafka/src/confluent_kafka.c:232: error: expected \u2018)\u2019 before \u2018err\u2019

confluent_kafka/src/confluent_kafka.c:258: error: expected \u2018)\u2019 before \u2018err\u2019

confluent_kafka/src/confluent_kafka.c: In function \u2018Message_partition\u2019:

confluent_kafka/src/confluent_kafka.c:312: error: \u2018RD_KAFKA_PARTITION_UA\u2019 undeclared (first use in this function)

confluent_kafka/src/confluent_kafka.c: In function \u2018Message_timestamp\u2019:

confluent_kafka/src/confluent_kafka.c:329: error: \u2018Message\u2019 has no member named \u2018tstype\u2019

confluent_kafka/src/confluent_kafka.c: At top level:

confluent_kafka/src/confluent_kafka.c:532: warning: type defaults to \u2018int\u2019 in declaration of \u2018rd_kafka_message_t\u2019

confluent_kafka/src/confluent_kafka.c:532: error: expected \u2018;\u2019, \u2018,\u2019 or \u2018)\u2019 before \u2018*\u2019 token

confluent_kafka/src/confluent_kafka.c:588: error: expected declaration specifiers or \u2018...\u2019 before \u2018rd_kafka_resp_err_t\u2019

confluent_kafka/src/confluent_kafka.c: In function \u2018TopicPartition_setup\u2019:

confluent_kafka/src/confluent_kafka.c:592: warning: implicit declaration of function \u2018KafkaError_new_or_None\u2019

confluent_kafka/src/confluent_kafka.c:592: error: \u2018err\u2019 undeclared (first use in this function)

confluent_kafka/src/confluent_kafka.c: In function \u2018TopicPartition_init\u2019:

confluent_kafka/src/confluent_kafka.c:608: error: \u2018RD_KAFKA_PARTITION_UA\u2019 undeclared (first use in this function)

confluent_kafka/src/confluent_kafka.c:609: error: \u2018RD_KAFKA_OFFSET_INVALID\u2019 undeclared (first use in this function)

confluent_kafka/src/confluent_kafka.c:620: error: too many arguments to function \u2018TopicPartition_setup\u2019

confluent_kafka/src/confluent_kafka.c: At top level:

confluent_kafka/src/confluent_kafka.c:797: error: expected declaration specifiers or \u2018...\u2019 before \u2018rd_kafka_resp_err_t\u2019

confluent_kafka/src/confluent_kafka.c: In function \u2018TopicPartition_new0\u2019:

confluent_kafka/src/confluent_kafka.c:803: error: \u2018err\u2019 undeclared (first use in this function)

confluent_kafka/src/confluent_kafka.c:803: error: too many arguments to function \u2018TopicPartition_setup\u2019

confluent_kafka/src/confluent_kafka.c: At top level:

confluent_kafka/src/confluent_kafka.c:816: warning: type defaults to \u2018int\u2019 in declaration of \u2018rd_kafka_topic_partition_list_t\u2019

confluent_kafka/src/confluent_kafka.c:816: error: expected \u2018;\u2019, \u2018,\u2019 or \u2018)\u2019 before \u2018*\u2019 token

confluent_kafka/src/confluent_kafka.c:839: error: expected \u2018=\u2019, \u2018,\u2019, \u2018;\u2019, \u2018asm\u2019 or \u2018__attribute__\u2019 before \u2018*\u2019 token

confluent_kafka/src/confluent_kafka.c:883: error: expected \u2018)\u2019 before \u2018*\u2019 token

confluent_kafka/src/confluent_kafka.c:909: error: expected \u2018)\u2019 before \u2018*\u2019 token

confluent_kafka/src/confluent_kafka.c: In function \u2018Handle_clear\u2019:

confluent_kafka/src/confluent_kafka.c:952: error: \u2018Handle\u2019 has no member named \u2018error_cb\u2019

confluent_kafka/src/confluent_kafka.c:953: error: \u2018Handle\u2019 has no member named \u2018error_cb\u2019

confluent_kafka/src/confluent_kafka.c:953: error: \u2018Handle\u2019 has no member named \u2018error_cb\u2019

confluent_kafka/src/confluent_kafka.c:953: error: \u2018Handle\u2019 has no member named \u2018error_cb\u2019

confluent_kafka/src/confluent_kafka.c:955: error: \u2018Handle\u2019 has no member named \u2018stats_cb\u2019

confluent_kafka/src/confluent_kafka.c:956: error: \u2018Handle\u2019 has no member named \u2018stats_cb\u2019

confluent_kafka/src/confluent_kafka.c:956: error: \u2018Handle\u2019 has no member named \u2018stats_cb\u2019

confluent_kafka/src/confluent_kafka.c:956: error: \u2018Handle\u2019 has no member named \u2018stats_cb\u2019

confluent_kafka/src/confluent_kafka.c:958: error: \u2018Handle\u2019 has no member named \u2018initiated\u2019

confluent_kafka/src/confluent_kafka.c:959: error: \u2018Handle\u2019 has no member named \u2018tlskey\u2019

confluent_kafka/src/confluent_kafka.c: In function \u2018Handle_traverse\u2019:

confluent_kafka/src/confluent_kafka.c:966: error: \u2018Handle\u2019 has no member named \u2018error_cb\u2019

confluent_kafka/src/confluent_kafka.c:967: error: \u2018Handle\u2019 has no member named \u2018error_cb\u2019

confluent_kafka/src/confluent_kafka.c:967: error: \u2018Handle\u2019 has no member named \u2018error_cb\u2019

confluent_kafka/src/confluent_kafka.c:969: error: \u2018Handle\u2019 has no member named \u2018stats_cb\u2019

confluent_kafka/src/confluent_kafka.c:970: error: \u2018Handle\u2019 has no member named \u2018stats_cb\u2019

confluent_kafka/src/confluent_kafka.c:970: error: \u2018Handle\u2019 has no member named \u2018stats_cb\u2019

confluent_kafka/src/confluent_kafka.c: At top level:

confluent_kafka/src/confluent_kafka.c:982: error: expected \u2018)\u2019 before \u2018*\u2019 token

confluent_kafka/src/confluent_kafka.c:1042: error: expected declaration specifiers or \u2018...\u2019 before \u2018rd_kafka_conf_t\u2019

confluent_kafka/src/confluent_kafka.c:1043: error: expected declaration specifiers or \u2018...\u2019 before \u2018rd_kafka_topic_conf_t\u2019

confluent_kafka/src/confluent_kafka.c: In function \u2018producer_conf_set_special\u2019:

confluent_kafka/src/confluent_kafka.c:1050: warning: implicit declaration of function \u2018KafkaError_new0\u2019

confluent_kafka/src/confluent_kafka.c:1050: error: \u2018RD_KAFKA_RESP_ERR__INVALID_ARG\u2019 undeclared (first use in this function)

confluent_kafka/src/confluent_kafka.c:1057: error: \u2018Handle\u2019 has no member named \u2018u\u2019

confluent_kafka/src/confluent_kafka.c:1058: error: \u2018Handle\u2019 has no member named \u2018u\u2019

confluent_kafka/src/confluent_kafka.c:1071: warning: implicit declaration of function \u2018rd_kafka_topic_conf_set_partitioner_cb\u2019

confluent_kafka/src/confluent_kafka.c:1072: error: \u2018tconf\u2019 undeclared (first use in this function)

confluent_kafka/src/confluent_kafka.c:1072: error: \u2018rd_kafka_msg_partitioner_random\u2019 undeclared (first use in this function)

confluent_kafka/src/confluent_kafka.c:1075: error: \u2018rd_kafka_msg_partitioner_consistent\u2019 undeclared (first use in this function)

confluent_kafka/src/confluent_kafka.c:1078: error: \u2018rd_kafka_msg_partitioner_consistent_random\u2019 undeclared (first use in this function)

confluent_kafka/src/confluent_kafka.c:1104: error: \u2018RD_KAFKA_RESP_ERR__NOT_IMPLEMENTED\u2019 undeclared (first use in this function)

confluent_kafka/src/confluent_kafka.c:1110: error: \u2018Handle\u2019 has no member named \u2018u\u2019

confluent_kafka/src/confluent_kafka.c:1111: error: \u2018Handle\u2019 has no member named \u2018u\u2019

confluent_kafka/src/confluent_kafka.c:1111: error: \u2018Handle\u2019 has no member named \u2018u\u2019

confluent_kafka/src/confluent_kafka.c:1111: error: \u2018Handle\u2019 has no member named \u2018u\u2019

confluent_kafka/src/confluent_kafka.c:1113: error: \u2018Handle\u2019 has no member named \u2018u\u2019

confluent_kafka/src/confluent_kafka.c:1114: error: \u2018Handle\u2019 has no member named \u2018u\u2019

confluent_kafka/src/confluent_kafka.c:1118: error: \u2018Producer_partitioner_cb\u2019 undeclared (first use in this function)

confluent_kafka/src/confluent_kafka.c: At top level:

confluent_kafka/src/confluent_kafka.c:1133: error: expected declaration specifiers or \u2018...\u2019 before \u2018rd_kafka_conf_t\u2019

confluent_kafka/src/confluent_kafka.c:1134: error: expected declaration specifiers or \u2018...\u2019 before \u2018rd_kafka_topic_conf_t\u2019

confluent_kafka/src/confluent_kafka.c: In function \u2018consumer_conf_set_special\u2019:

confluent_kafka/src/confluent_kafka.c:1139: error: \u2018RD_KAFKA_RESP_ERR__INVALID_ARG\u2019 undeclared (first use in this function)

confluent_kafka/src/confluent_kafka.c:1146: error: \u2018Handle\u2019 has no member named \u2018u\u2019

confluent_kafka/src/confluent_kafka.c:1147: error: \u2018Handle\u2019 has no member named \u2018u\u2019

confluent_kafka/src/confluent_kafka.c: At top level:

confluent_kafka/src/confluent_kafka.c:1162: error: expected \u2018=\u2019, \u2018,\u2019, \u2018;\u2019, \u2018asm\u2019 or \u2018__attribute__\u2019 before \u2018*\u2019 token

confluent_kafka/src/confluent_kafka.c: In function \u2018CallState_begin\u2019:

confluent_kafka/src/confluent_kafka.c:1335: error: \u2018Handle\u2019 has no member named \u2018tlskey\u2019

confluent_kafka/src/confluent_kafka.c: In function \u2018CallState_end\u2019:

confluent_kafka/src/confluent_kafka.c:1343: error: \u2018Handle\u2019 has no member named \u2018tlskey\u2019

confluent_kafka/src/confluent_kafka.c: In function \u2018CallState_get\u2019:

confluent_kafka/src/confluent_kafka.c:1358: error: \u2018Handle\u2019 has no member named \u2018tlskey\u2019

confluent_kafka/src/confluent_kafka.c: In function \u2018libversion\u2019:

confluent_kafka/src/confluent_kafka.c:1396: warning: implicit declaration of function \u2018rd_kafka_version_str\u2019

confluent_kafka/src/confluent_kafka.c:1397: warning: implicit declaration of function \u2018rd_kafka_version\u2019

confluent_kafka/src/confluent_kafka.c: In function \u2018KafkaError_add_errs\u2019:

confluent_kafka/src/confluent_kafka.c:1438: warning: implicit declaration of function \u2018rd_kafka_get_err_descs\u2019

confluent_kafka/src/confluent_kafka.c:1474: error: invalid use of undefined type \u2018struct rd_kafka_err_desc\u2019

confluent_kafka/src/confluent_kafka.c:1474: error: dereferencing pointer to incomplete type

confluent_kafka/src/confluent_kafka.c:1477: error: invalid use of undefined type \u2018struct rd_kafka_err_desc\u2019

confluent_kafka/src/confluent_kafka.c:1477: error: dereferencing pointer to incomplete type

confluent_kafka/src/confluent_kafka.c:1479: error: invalid use of undefined type \u2018struct rd_kafka_err_desc\u2019

confluent_kafka/src/confluent_kafka.c:1479: error: dereferencing pointer to incomplete type

confluent_kafka/src/confluent_kafka.c:1483: error: invalid use of undefined type \u2018struct rd_kafka_err_desc\u2019

confluent_kafka/src/confluent_kafka.c:1483: error: dereferencing pointer to incomplete type

confluent_kafka/src/confluent_kafka.c:1483: error: invalid use of undefined type \u2018struct rd_kafka_err_desc\u2019

confluent_kafka/src/confluent_kafka.c:1483: error: dereferencing pointer to incomplete type

confluent_kafka/src/confluent_kafka.c: In function \u2018_init_cimpl\u2019:

confluent_kafka/src/confluent_kafka.c:1561: error: \u2018RD_KAFKA_TIMESTAMP_NOT_AVAILABLE\u2019 undeclared (first use in this function)

confluent_kafka/src/confluent_kafka.c:1562: error: \u2018RD_KAFKA_TIMESTAMP_CREATE_TIME\u2019 undeclared (first use in this function)

confluent_kafka/src/confluent_kafka.c:1563: error: \u2018RD_KAFKA_TIMESTAMP_LOG_APPEND_TIME\u2019 undeclared (first use in this function)

confluent_kafka/src/confluent_kafka.c:1565: error: \u2018RD_KAFKA_OFFSET_BEGINNING\u2019 undeclared (first use in this function)

confluent_kafka/src/confluent_kafka.c:1566: error: \u2018RD_KAFKA_OFFSET_END\u2019 undeclared (first use in this function)

confluent_kafka/src/confluent_kafka.c:1567: error: \u2018RD_KAFKA_OFFSET_STORED\u2019 undeclared (first use in this function)

confluent_kafka/src/confluent_kafka.c:1568: error: \u2018RD_KAFKA_OFFSET_INVALID\u2019 undeclared (first use in this function)

error: command 'gcc' failed with exit status 1

----------------------------------------
Cleaning up...
Command /home/eadeegf/tmp/python/bin/python -c "import setuptools, tokenize;__file__='/tmp/pip_build_eadeegf/confluent-kafka/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-zJROpL-record/install-record.txt --single-version-externally-managed --compile failed with error code 1 in /tmp/pip_build_eadeegf/confluent-kafka
Traceback (most recent call last):
  File "/home/eadeegf/tmp/python/bin//pip", line 9, in <module>
    load_entry_point('pip==1.5.4', 'console_scripts', 'pip')()
  File "/home/eadeegf/tmp/python/lib/python2.7/site-packages/pip-1.5.4-py2.7.egg/pip/__init__.py", line 185, in main
    return command.main(cmd_args)
  File "/home/eadeegf/tmp/python/lib/python2.7/site-packages/pip-1.5.4-py2.7.egg/pip/basecommand.py", line 161, in main
    text = '\n'.join(complete_log)
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 62: ordinal not in range(128)

**Comment by edenhill** [comment]
Seems like it can't find the librdkafka-dev development files.
How did you install librdkafka?
I suggest using one of the prebuilt packages as instructed here:
https://github.com/confluentinc/confluent-kafka-python#prerequisites

(also try to upgrade librdkafka to v0.9.4 or later, v0.9.1 is quite old now)

**Comment by 1182986209** [comment]
I download the v0.9.5.tar.gz.
tar -zxvf v0.9.5.tar.gz
cd librdkafka
./configure --prefix=/home/eadeegf/tmp/librdkafka
make && make install

**Comment by edenhill** [comment]
Since /home/eadeegf/tmp/librdkafka is not a standard location there is no way for the Python to find it.
Either specify this path (for both include and lib) for the build process, and later the lib path every time you want to run the client, or .. much better, install librdkafka to a standard location, e.g., /usr, by: `./configure --prefix=/usr && make && sudo make install`

**Comment by 1182986209** [comment]
o, still have problem. How could I specify the  path of my own for the build process? I can not got permission  to run sudo.

**Comment by edenhill** [comment]
See this SO answer:
https://stackoverflow.com/questions/6928110/how-may-i-override-the-compiler-gcc-flags-that-setup-py-uses-by-default

You'll want to set `CFLAGS=-I/path/to/where/you/installed/librdkafka/include` and `-L/path/to/where/you/installed/librdkafka/lib`, but you will also need to pass `LD_LIBRARY_PATH=/path/to/where/you/installed/librdkafka/lib` for each invocation of your program to make it find the library.  This is why installing librdkafka properly at the system level is advised.

**Comment by PabloLefort** [comment]
Running on docker also! 🚀 
```Docker
FROM python:3.7.0-alpine3.8
ENV PYTHONUNBUFFERED 1
RUN mkdir -p /code
RUN apk update && apk add bash gcc librdkafka-dev python3-dev
WORKDIR /code
ADD . /code/
RUN pip install -r requirements.txt
...
```
